Keygen.php

```diff
@@ -42,17 +41,7 @@ private static function currentSaveToInt(int $currentSave) {
 	 * ex. '0' = 0, '1' = 1, '2' = 2, 'a' = 1, 'b' = 2
 	 */
     private static function charToInt(string $char) : int {
-		if($char === '0') {
-			return 0;
-		}
-
-		$num = intval($char);
-
-		if($num === 0) {
-			$num = ord($char) - 96;
-		}
-
-		return $num;
+		return intval($char) + (ord($char) - 96) * ((intval($char) === 0) && ($char !== '0'));
```

Comment:
This makes the code quite a bit less readable but if your other changes are fine and the profileIds are still made correctly i'll use the other changes (not sure if you tested before making pr)